dev/Dockerfile

```diff
@@ -18,88 +19,109 @@ RUN cargo chef prepare --recipe-path recipe.json
 # provides ARM binaries: https://github.com/LukeChannings/deno-arm64
 FROM lukechannings/deno:v${DENO_VERSION} AS deno-bin
 
-#
 FROM base AS builder
 
-ARG WASMEDGE_VERSION=0.13.5
-
 ENV DENO_DIR /deno-dir/
 ENV DENO_INSTALL /root/.deno
 ENV PATH "${DENO_INSTALL}/bin:${PATH}"
 
 COPY --from=deno-bin /usr/bin/deno /bin/deno
 
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-    # base
-    git \
-    curl \
-    python3 \
-    make \
-    cmake \
-    # grpc
-    protobuf-compiler \
-    libprotobuf-dev \
-    # openssl
-    pkg-config \
-    libssl-dev \
-    # wasmedge
-    libclang-dev \
-    && rm -rf /var/lib/apt/lists/* \
-    && curl -fsS https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh -o wasmedge.sh \
-    && bash wasmedge.sh -v $WASMEDGE_VERSION \
-    && rm wasmedge.sh
+RUN set -eux; \
+    export DEBIAN_FRONTEND=noninteractive; \
+    apt update; \
+    apt install --yes --no-install-recommends \
+        # typegate build \
+        make \
+        ## libffi-sys\
+        automake \
+        ## wasmedge-sys \
+        libclang-dev \
+        ## openssl deps \
+        pkg-config \
+        libssl-dev \
+        # base ghjk deps \
+        git \
+        curl \
+        # asdf deps \
+        xz-utils \
+        unzip \
+        ; \
+    apt clean autoclean; apt autoremove --yes; rm -rf /var/lib/{apt,dpkg,cache,log}/;
+
+ARG GHJK_VERSION=6040bb3
+RUN GHJK_INSTALL_EXE_DIR=/usr/bin GHJK_INSTALL_HOOK_SHELLS=bash deno run -A https://raw.github.com/metatypedev/ghjk/$GHJK_VERSION/install.ts
+
+COPY ghjk.ts .
+RUN CI=1 NO_PYTHON=1 ghjk ports sync
+ENV GHJK_ENV=/root/.local/share/ghjk/env.sh
 
 COPY --from=plan /app/recipe.json recipe.json
 
-RUN cargo chef cook --recipe-path recipe.json --release --package typegate \
-    && rm recipe.json
+RUN <<EOT
+    . $GHJK_ENV
+    cargo chef cook --recipe-path recipe.json --release --package typegate
+    rm recipe.json
+EOT
 
 COPY . .
 
-RUN cargo build --release --package typegate
+RUN <<EOT
```

Comment:
with EOT, should not set -eu be set each time? if then, not sure the syntax is cleaner than the chaining

Reply:
Oh yeah. I keep forgetting that, will fix.